src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -763,6 +764,15 @@ uct_ib_mlx5_devx_memh_alloc(uct_ib_mlx5_md_t *md, size_t length,
     return UCS_OK;
 }
 
+static int uct_ib_mlx5_devx_memh_has_ro(uct_ib_mlx5_md_t *md,
+                                        uct_ib_mlx5_devx_mem_t *memh) {
+    if (memh->mrs[UCT_IB_MR_DEFAULT].super.ib->addr != NULL) {
+        return md->super.relaxed_order;
+    }
+
+    return (md->flags & UCT_IB_MLX5_MD_FLAG_GVA_RO) && (md->super.relaxed_order);
```

Comment:
i'd suggest to not set `UCT_IB_MLX5_MD_FLAG_GVA_RO` flag at all if `md->super.relaxed_order` is false